indico/modules/events/editing/client/js/management/editable_type/EditableList.jsx

```diff
@@ -158,36 +206,114 @@ function EditableListDisplay({editableList, codePresent, editableType, eventId})
 
   const toggleSelectAll = dataChecked => {
     if (dataChecked) {
-      setChecked(sortedList.map(row => row.id));
+      setChecked(editables.map(row => row.editable.id));
     } else {
       setChecked([]);
     }
   };
 
   const toggleSelectRow = dataIndex => {
-    const newRow = sortedList[dataIndex].id;
+    const newRow = sortedList[dataIndex].editable.id;
     if (checked.includes(newRow)) {
       setChecked(old => old.filter(row => row !== newRow));
     } else {
       setChecked(old => [...old, newRow]);
     }
   };
 
+  const checkedEditablesRequest = async (urlFunc, data = {}) => {
+    let response;
+    try {
+      response = await indicoAxios.post(urlFunc({confId: eventId, type: editableType}), {
+        editables: checked,
+        ...data,
+      });
+    } catch (error) {
+      handleAxiosError(error);
+      return null;
+    } finally {
+      setActiveRequest(null);
+    }
+
+    return camelizeKeys(response.data);
+  };
+
+  const downloadAllFiles = async () => {
+    setActiveRequest('download');
+    const rv = await checkedEditablesRequest(editablesArchiveURL);
+    if (rv) {
+      location.href = rv.downloadURL;
+    }
+  };
+
+  const assignEditor = async editor => {
+    setActiveRequest('assign');
+    const rv = await checkedEditablesRequest(assignEditorURL, {editor});
+    if (rv) {
+      patchList(rv);
+    }
+  };
+
+  const assignSelfEditor = async () => {
+    setActiveRequest('assign-self');
+    const rv = await checkedEditablesRequest(assignSelfEditorURL);
+    if (rv) {
+      patchList(rv);
+    }
+  };
+
+  const unassignEditor = async () => {
+    setActiveRequest('unassign');
+    const rv = await checkedEditablesRequest(unassignEditorURL);
+    if (rv) {
+      patchList(rv);
+    }
+  };
+
   return (
     <>
       <ManagementPageSubTitle title={title} />
-      <ManagementPageBackButton url={dashboardURL({confId: eventId})} />
+      <ManagementPageBackButton url={editableTypeURL({confId: eventId, type: editableType})} />
       <div styleName="editable-topbar">
         <div>
-          <Button disabled={hasCheckedRows} content={Translate.string('Assign')} />
-          <Button disabled={hasCheckedRows} content={Translate.string('Unassign')} />
-          <Button disabled={hasCheckedRows} content={Translate.string('Set status')} />
+          <Button.Group>
+            <Dropdown
```

Comment:
The assignments look really nice! But I noticed that for a high number of editors the dropdown will become very large and sometimes also opening outside of the margins.
I'm wondering if we could give it a fixed size and make it scrollable? 
A "search" function would also look dope but I'm featurizing now 😄

![image](https://user-images.githubusercontent.com/12183954/84658348-a7a0f280-af0d-11ea-8d88-8e5524cb4874.png)

Reply:
adding `scrolling` in the dropdown already avoided this ugliness :)